rtloader/three/three.cpp

```diff
@@ -34,19 +34,50 @@ extern "C" DATADOG_AGENT_RTLOADER_API void destroy(RtLoader *p)
 
 Three::Three(const char *python_home, const char *python_exe, cb_memory_tracker_t memtrack_cb)
     : RtLoader(memtrack_cb)
-    , _pythonHome(NULL)
-    , _pythonExe(NULL)
     , _baseClass(NULL)
     , _pythonPaths()
     , _pymallocPrev{ 0 }
     , _pymemInuse(0)
     , _pymemAlloc(0)
 {
-    initPythonHome(python_home);
+    // Pre-Initialize Python with UTF-8 mode
+    PyStatus status;
+    PyPreConfig preconfig;
+    PyPreConfig_InitIsolatedConfig(&preconfig);
+    preconfig.utf8_mode = 1;
+
+    status = Py_PreInitialize(&preconfig);
+    if (PyStatus_Exception(status)) {
+        setError("Failed to pre-initialize Python: " + std::string(status.err_msg));
```

Comment:
I don't think anything checks for an error set this weay when we call the constructor. Would it be possible to add a test for this code path?